src/main/java/net/dv8tion/jda/webhook/WebhookUser.java

```diff
@@ -0,0 +1,168 @@
+/*
+ *     Copyright 2015-2018 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.webhook;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.entities.Guild;
+import net.dv8tion.jda.core.entities.PrivateChannel;
+import net.dv8tion.jda.core.entities.User;
+import net.dv8tion.jda.core.entities.impl.UserImpl.DefaultAvatar;
+import net.dv8tion.jda.core.requests.RestAction;
+
+import java.util.List;
+
+/**
+ * Represents a user from a webhook, this can be a user that was mentioned or the webhook user itself
+ */
+public class WebhookUser implements User
```

Comment:
I'm concerned about this. We use the `User` interface throughout JDA expecting its always implemented by `UserImpl`. This would mean we might cast it without checking types assuming that no error will be thrown. The documentation does not make it clear that JDA related methods such as `getMutualGuilds` will throw an exception.